atef/widgets/config/page.py

```diff
@@ -410,44 +321,42 @@ def setup_parent_button(self, button: QToolButton) -> None:
         # Make sure the button's starting tooltip is correct
         self.update_parent_tooltip()
 
-    def setup_child_button(self, button: QToolButton, item: AtefItem) -> None:
+    def setup_child_button(self, button: QToolButton, item: TreeItem) -> None:
         """
         Set up a button's style and make it navigate to a specific child page.
         """
-        # Create a function that navigates to the right page
-        def inner_navigate(*args, **kwargs):
-            self.navigate_to(item)
+        navigate_slot = WeakPartialMethodSlot(button, button.clicked,
+                                              self.navigate_to, item)
+        self._partial_slots.append(navigate_slot)
```

Comment:
If we expect there to be a lot of these, we might consider pruning the list occasionally (i.e., removing those that have been unsubscribed)

Reply:
Or perhaps better: `WeakPartialMethodSlot` could become a `QObject`, emit a "Destroyed!" signal when any goes out of scope - then `_partial_slots` could be cleaned automatically... Hmmm